app/assets/javascripts/character_count.js

```diff
@@ -4,57 +4,60 @@ $(() => {
     el.removeClass(icons.join(' ')).addClass(icon);
   };
 
-  $(document).on('keyup change paste', '[data-character-count]', ev => {
+  const setCounterState = (el, state) => {
+    if(state == 'info') {
+      el.removeClass('has-color-yellow-700 has-color-red-500').addClass('has-color-primary');
+    } else if(state === 'warning') {
+      el.removeClass('has-color-red-500 has-color-primary').addClass('has-color-yellow-700');
+    } else if(state === 'error') {
+      el.removeClass('has-color-yellow-700 has-color-primary').addClass('has-color-red-500');
+    } else {
+      el.removeClass('has-color-red-500 has-color-yellow-700 has-color-primary');
+    }
+  }
+
+  const setInputState = (el, state) => {
+    el.toggleClass('failed-validation', state);
+  };
+
+  const setSubmitButtonState = (el, state) => {
+    if (el) {
+      el.attr('disabled', state).toggleClass('is-muted', state);
+    }
+  };
+
+  $(document).on('keyup change paste', '[data-character-count]', (ev) => {
     const $tgt = $(ev.target);
     const $counter = $($tgt.attr('data-character-count'));
     const $button = $counter.parents('form').find('input[type="submit"]');
     const $count = $counter.find('.js-character-count__count');
     const $icon = $counter.find('.js-character-count__icon');
 
-    const displayAt = parseFloat($counter.attr('data-display-at'));
+    const count = $tgt.val().length;
     const max = parseInt($counter.attr('data-max'), 10);
     const min = parseInt($counter.attr('data-min'), 10);
-    const count = $tgt.val().length;
-    const text = `${count} / ${max}`;
+    const threshold = parseFloat($counter.attr('data-threshold'));
 
-    if (displayAt) {
-      if (count >= displayAt * max) {
-        $counter.removeClass('hide');
-      }
-      else {
-        $counter.addClass('hide');
-      }
-    }
+    const gtnMax = count > max;
+    const ltnMin = count < min;
+    const gteThreshold = count >= threshold * max;
 
-    if (count > max) {
-      $counter.removeClass('has-color-yellow-700 has-color-primary').addClass('has-color-red-500');
+    const text = `${count} / ${ltnMin ? min : max}`;
+    
+    if(gtnMax || ltnMin) {
+      setCounterState($counter, 'error');
       setIcon($icon, 'fa-times');
-      if ($button) {
-        $button.attr('disabled', true).addClass('is-muted');
-      }
-    }
-    else if (count > 0.75 * max) {
-      $counter.removeClass('has-color-red-500 has-color-primary').addClass('has-color-yellow-700');
+      setSubmitButtonState($button, true);
```

Comment:
comment: I had to double check the implementation of `setSubmitButtonState` here - passing `true` when you want to disable the input feels weird. Makes sense when you actually look, but might be worth doing something about. Possibilities:

 * Rename the `state` parameter in each of these functions to `isDisabled`, or similar
 * `setSubmitButtonState($button, { disabled: true })` (might be overkill)
 * Invert the parameter in each function `const disabled = !state;`

Or something else!

Reply:
Upon further inspection, I think the state can simply be a union of string literals `'enabled' | 'disabled'` with a `state === 'disabled'` check under the hood. Should make the intent clear as day (I don't like boolean / magic number parameters either)